### PR TITLE
Customizable Rollup Recalculation batch job fails if no Allocations in a given batch look up to an Opportunity record

### PR DESCRIPTION
**Pull request by screcco-sfdo**
[W-11174448](https://gus.lightning.force.com/a07EE00000xOfH5YAK)

# Critical Changes

# Changes

# Issues Closed
- [Known Issue](https://trailblazer.salesforce.com/issues_view?id=a1p4V000002dlXiQAI): Customizable Rollups: List index out of bounds error is thrown if GAU recalculation batch does not contain Opportunity Allocations
# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: force-app/main/default/classes/CRLP_Batch_Base_Skew.cls [hunk 1]
```diff
@@ -232,7 +232,7 @@ public abstract class CRLP_Batch_Base_Skew extends CRLP_Batch_Base {
     * @param parentObjectField
     * @param detailRecords
     * @param parentKeyField
-    * @return Boolean true if lastParentId has no associated detail records in a GAU rollup
+    * @return Boolean false if lastParentId has no associated detail records in a GAU rollup
```
Comment (lparrott) [inline force-app/main/default/classes/CRLP_Batch_Base_Skew.cls:235]: Method documentation is more useful when it's correct, eh? 🙃

File: force-app/main/default/classes/CRLP_GAU_BATCH_TEST.cls [hunk 2]
```diff
@@ -57,6 +57,23 @@ private with sharing class CRLP_GAU_BATCH_TEST {
         System.assertEquals(false,
             gauBatch.hasLastParentDetails(lastParentId, null, detailRecords, parentKeyField));
     }
+    
+    @IsTest
+    private static void shouldDetermineLastParentIdHasNoDetailsEmptyDetailRecords() {
```
Comment (lparrott) [inline force-app/main/default/classes/CRLP_GAU_BATCH_TEST.cls:62]: Nice job on the Unit test, it only does what it needs to do 👍

**Review by lparrott — APPROVED** [review]

File: force-app/main/default/classes/CRLP_Batch_Base_Skew.cls (hunk 1 above)
Comment (screcco-sfdo) [reply force-app/main/default/classes/CRLP_Batch_Base_Skew.cls:235]: documentation is like 🍕 even if it's not great, it's still pretty good

**Review by voduyemi — APPROVED** [review]
Looks good to me! Thanks @screcco-sfdo

**Pull request merged**